include/revng/Support/IRHelpers.h

```diff
@@ -984,6 +984,25 @@ inline std::string dumpToString(T &TheT) {
   return Result;
 }
 
+// clang-format off
+template<typename T>
+concept LLVMRawOStreamDumpable = not ValueLikePrintable<T>
+                                 and not ModFunLikePrintable<T>
+                                 and requires(T TheT) {
+  TheT.dump(std::declval<llvm::raw_ostream &>());
+};
+// clang-format on
+
+template<bool B, LLVMRawOStreamDumpable Dumpable>
+static void writeToLog(Logger<B> &L, const Dumpable &P, int /* Ignore */) {
+  if (L.isEnabled()) {
+    llvm::SmallString<32> Buffer;
+    llvm::raw_svector_ostream Stream(Buffer);
+    P.dump(Stream);
```

Comment:
I usually do like this in order to be sure that the stream is flushed, but it's probably not necessary.
```suggestion
    {
      llvm::raw_svector_ostream Stream(Buffer);
      P.dump(Stream);
    }
```